File: solutions/python/rotational-cipher/1/rotational_cipher.py
def rotate_char(char: str, key: int) -> str:
    if char.isspace() or not char.isalpha():
        return char
    offset = ord('A') if char.isupper() else ord('a')
    return chr(((ord(char) - offset + key) % 26) + offset)


def rotate(text: str, key: int) -> str:
    return ''.join(rotate_char(char, key) for char in text)

File: solutions/ruby/rotational-cipher/1/rotational_cipher.rb
module RotationalCipher
  def self.shift(char, key)
    return char if char =~ /[^A-Za-z]/

    char_ord = char =~ /[A-Z]/ ? 'A'.ord : 'a'.ord

    (((char.ord - char_ord + key) % 26) + char_ord).chr
  end

  def self.rotate(plaintext, key)
    plaintext.chars.map{|char| shift(char, key)}.join
  end
end
